client.go

```diff
@@ -925,12 +935,40 @@ func anyError(resp *http.Response, err error) error {
 }
 
 // doRequest creates and performs an http request.
-func (c *Client) doRequest(host *URI, method, path string, headers map[string]string, reader io.Reader) (*http.Response, error) {
-	req, err := makeRequest(host, method, path, headers, reader)
+func (c *Client) doRequest(host *URI, method, path string, headers map[string]string, data []byte) (*http.Response, error) {
+	var resp *http.Response
+	var err error
+	var req *http.Request
+	retrial := 1 + c.retrials
+	sleepTime := c.minRetrySleepTime
+	for retrial > 0 {
+		req, err = makeRequest(host, method, path, headers, data)
+		if err != nil {
+			return nil, errors.Wrap(err, "building request")
+		}
+		retrial -= 1
```

Comment:
i think one of the linters will ask for "--" here.

Reply:
updated.